File: zd3.rs
use std::io;

fn is_prime(n: i32) -> bool {
    if n <= 1 {
        return false;
    }
    for i in 2..=(n as f64).sqrt() as i32 {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let mut numbers: Vec<i32> = Vec::new();
    println!("Введите числа (для завершения введите отрицательное число):");
    
    loop {
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let num: i32 = match input.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                eprintln!("Нужно вводить только числа:");
                continue;
            }
        };
        
        if num < 0 {
            break;
        }
        
        numbers.push(num);
    }
    
    println!("Результаты проверки чисел на простоту:");
    for number in numbers {
        if is_prime(number) {
            println!("{} - простое число", number);
        } else {
            println!("{} - не является простым числом", number);
        }
    }
}

File: ZD3.swift
import Foundation

func isPrime(_ n: Int) -> Bool {
    if n <= 1 {
        return false
    }
    for i in 2...Int(sqrt(Double(n))) {
        if n % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    var numbers = [Int]()
    var num: Int = 0
    
    print("Введите числа (для завершения введите отрицательное число):")
    while true {
        if num >= 0 {
            if let input = readLine(), let inputNum = Int(input) {
                num = inputNum
                if num < 0 {
                    break
                }
                numbers.append(num)
            }
        } else {
            print("Нужно вводить только числа:")
            return
        }
    }
    
    print("Результаты проверки чисел на простоту:")
    for number in numbers {
        if isPrime(number) {
            print("\(number) - простое число")
        } else {
            print("\(number) - не является простым числом")
        }
    }
}

main()
